### PR TITLE
Vb 2090 update sql generator for updated category grouping

### PR DESCRIPTION
**Pull request by dhirajshettymoj**
## What does this pull request do?

Updates SQL generator to support grouping by category.
## What is the intent behind these changes?

Updates SQL generator to support grouping by category and make code more generic.

### CODE REVIEW

**Review by AledWEvans — APPROVED** [review]
LGTM

File: src/main/resources/session-template-data/session-data.csv [hunk 1]
```diff
@@ -1,165 +1,165 @@
...
-MHI,Visits Main Hall,SOCIAL,14,2,FALSE,13:15:00,16:00:00,2023-04-07,,THURSDAY,FALSE,,,
+MHI,Visits Main Hall,SOCIAL,14,2,FALSE,13:15:00,16:00:00,2023-04-07,,THURSDAY,FALSE,,
 % BNI Prison
-BNI,Visits Main Hall,SOCIAL,60,5,FALSE,14:15:00,15:45:00,2023-03-10,,TUESDAY,FALSE,,,
-BNI,Visits Main Hall,SOCIAL,60,5,FALSE,14:15:00,15:45:00,2023-03-10,,WEDNESDAY,FALSE,,,
-BNI,Visits Main Hall,SOCIAL,60,5,FALSE,14:15:00,15:45:00,2023-03-10,,THURSDAY,FALSE,,,
-BNI,Visits Main Hall,SOCIAL,60,5,FALSE,14:15:00,15:45:00,2023-03-10,,SATURDAY,FALSE,,,
-BNI,Visits Main Hall,SOCIAL,60,5,FALSE,14:15:00,15:45:00,2023-03-10,,SUNDAY,FALSE,,,
\ No newline at end of file
+BNI,Visits Main Hall,SOCIAL,60,5,FALSE,14:15:00,15:45:00,2023-03-10,,TUESDAY,FALSE,,
```
Comment (AledWEvans) [inline src/main/resources/session-template-data/session-data.csv:161]: not part of this review but please update time to new BNI values..

**Review by AledWEvans — CHANGES_REQUESTED** [review]

File: src/main/resources/session-template-data/session-data.csv (hunk 1 above)
Comment (dhirajshettymoj) [reply src/main/resources/session-template-data/session-data.csv:161]: yes, need to do that,  will do that as part of a separate PR.

Comment (AledWEvans) [reply src/main/resources/session-template-data/session-data.csv:161]: okay will approve with that in mind

**Review by AledWEvans — APPROVED** [review]
LGTM

**Pull request merged**